.github/workflows/release-marketing-app.yml

```diff
@@ -70,3 +77,45 @@ jobs:
           subject-name: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME}}
           subject-digest: ${{ steps.push.outputs.digest }}
           push-to-registry: true
+
+  # Job: deploy-to-marketing-test
+  # This job pulls the task definition for the current task running in the cluster and updates the docker image tag with the one built in the previous job
+  # Then, the task definition is deployed to the ECS service
+  # See: https://docs.github.com/en/actions/use-cases-and-examples/deploying/deploying-to-amazon-elastic-container-service
+  deploy-to-marketing-test:
+    runs-on: ubuntu-24.04
+    environment: marketing-test
+    needs: build-and-push-image
+
+    steps:
+    # Warning: Use minimal permissions for the deployer
+    # See: https://github.com/aws-actions/amazon-ecs-deploy-task-definition?tab=readme-ov-file#permissions
+    - name: Configure ECS/Github Deployer AWS credentials
+      uses: aws-actions/configure-aws-credentials@v4.0.2
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: ${{ vars.AWS_REGION }}
+
+    # Retrieves the latest task definition in ECS and downloads it locally.
+    - name: Download latest task definition
+      run: |
+        aws ecs describe-task-definition --task-definition ${{ vars.ECS_TASK_DEFINITION_ARN }} --query taskDefinition > task-definition.json
+
+    # This updates the task definition from the previous state with the docker image produced by the build-and-push-image job.
+    - name: Fill in the new image ID in the Amazon ECS task definition
+      id: task-def
+      uses: aws-actions/amazon-ecs-render-task-definition@v1.6.2
+      with:
+        task-definition: task-definition.json
+        container-name: web
+        image: ${{ needs.build-and-push-image.outputs.fullImageUri }}
+
+    # Deploys the task definition generated in the previous step to ECS
+    - name: Deploy Amazon ECS task definition
+      uses: aws-actions/amazon-ecs-deploy-task-definition@v2.2.0
+      with:
+        task-definition: ${{ steps.task-def.outputs.task-definition }}
+        service: ${{ vars.ECS_SERVICE }}
+        cluster: ${{ vars.ECS_CLUSTER }}
+        wait-for-service-stability: true
```

Comment:
If the ECS Service (and it's Tasks) are deployed as cloudformation stacks, we should update the task definition via a cloudformation update, not by manually editing the task definition.

I'd recommend having this github action perform a cloudformation deploy, and pass the image id as a Parameter.